### DISCUSSION TITLE
Is it possible to comment on dependabot alerts?

### DISCUSSION THREAD

**Discussion by carlgieringer** [community/community#32674] [Code Security]
I started looking into a dependabot alert, and it will require upgrading a core dependency. Upgrading will be time consuming and will require careful coordination. I am not able to perform the ugprade now, but I would like to comment on the dependabot alert with my research findings and ideas. I would also like to block the dependabot alert on an issue dedicated to upgrading that core dependency (the upgrade may itself have blockers, and there will be several PRs and discussions related to the upgrade.)

Is it possible to comment on dependabot alerts? Another possible way to handle this would be to 'create an issue from a dependabot alert'. The dependabot alert could provide a link to the issue for later discovery, and the newly created issue would be created from a template linking to the dependabot alert. Then all the comments, discussion, PRs, blockers, etc. could go into that newly created issue.

**Comment by samigt** [comment]
In the same line of thought, an option to forward the dependabot alerts to the security team for review

**Comment by naomi789** [comment]
Great feedback, thank you! Unfortunately, my understanding is that folks can currently only comment for dismissals.

**Comment by jornncc** [comment]
What I end up doing is literally to 'dismiss the alert --> provide rally/jira ticket as a comment' and then re-open the ticket again :-) That way the comment shows up on the ticket.

**Comment by amaciejk** [comment]
Another vote for this - it would be really useful for alerts where the investigation has been done and you're just waiting for a new release of a dependancy or the like

**Comment by lukewhitehouse** [comment]
Bumping this one. I'm on a mission to resolve dependabot alerts across many legacy internal repos at my company and it's making me feel like Loki (s2 spoilers) with all the dependency trees I'm having to weave together.

Being able to comment on or even assign an alert for other contributors to see would be amazing for transparency and collaboration.

**Comment by kevin-aqqo** [comment]
Bumped. Would allow for easier identification of [planned/ongoing issues] vs [new issues] without having to 'resolve/dismiss' them before they're actually fixed.

**Comment by robinbristoluk** [comment]
I agree this would be a very valuable feature, allowing teams to keep track on issues

**Comment by supriyaalisetty501** [comment]
Hello — at present, Dependabot alerts do not support general threaded comments in the same way as Issues or Pull Requests.

Based on the current functionality, comments are typically available only when dismissing an alert, where you can provide a reason or additional context. Standard collaboration features such as ongoing discussion, assignment, or progress tracking are limited within the alert itself.

As a practical workaround, many teams use one of the following approaches:

1. Create a dedicated Issue linked to the alert
   Use the Issue to track investigation notes, blockers, related pull requests, and upgrade planning.

2. Reference an internal ticket
   Link a Jira, Azure DevOps, or other tracking ticket in the dismissal note or internal documentation.

3. Reopen alerts when work resumes
   If temporarily dismissed for tracking purposes, the alert can be revisited later when remediation is scheduled.

4. Use Security Overview or project boards
   For organizations, broader tracking can be managed through internal workflows and project management tools.

Your suggestion to allow direct comments or create an Issue from a Dependabot alert would be a valuable improvement for collaboration, transparency, and long-term remediation planning.

**Comment by schizek** [comment]
+1 from me as well... our team would benefit from this.
